specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,983 @@
+program output. The Base Profile specification hence requires explicitly
+expressing which values/measurements are returned by the program and in which
+order. How to express this is defined in the section on [output
+recording](#output-recording).
+
+While it is sufficient for the QPU to do a final measurement of all qubits in a
+predefined order at the end of the program, only the selected subset will be
+reflected in the produced output schema. A suitable output schema can be 
+generated in a post-processing step after the computation on the quantum
+processor itself has completed; customization of the program output hence does
+not require support on the QPU itself.
+
+The defined [output schemas](../output_schemas/) provide different options for
+how a backend may express the computed value(s). The exact schema can be freely
+chosen by the backend and is identified by a string label in the produced
+schema. Each output schema contains sufficient information to allow quantum
+programming frameworks to generate a user-friendly presentation of the returned
+values in the requested order, such as, e.g., a histogram of all results when
+running the program multiple times.
+
+**Bullet 4: Conditional Forward branching** <br />
+The Adaptive Profile can allow for arbitrary forward branching with the restriction being that
+branch instructions cannot express programs that do not terminate. By default, support for
+backwards branching is not a requirement, though a back-end can opt into **Bullet 13** to support
+backwards branching.
+Moreover, it's allowable that gates are conditionally performed based on the
+control flow to various blocks based on ** Bullet 6**.
+Depending on the vector of capabilities defined by an Adaptive Profile program, 
+proving non-termination with a static analysis may impossible due to the capabilities
+forming a Turing-complete subset of LLVM. Thus, it is up to back-ends to enforce 
+termination guarantees via a means of their choosing (for example a watchdog process
+with a timeout that will kill an executing Adaptive Profile program if it takes
+too much time). The following forms of branch instructions can be supported in the 
+profile, but at least the basic `br` instruction must be supported from the list below :
+| LLVM Instruction         | Context and Purpose                                                                              | Rules for Usage                                                                                             |
+| :----------------------- | :----------------------------------------------------------------------------------------------- | :---------------------------------------------------------------------------------------------------------- |
+| `br i1 label %true_branch, label %false_branch`                   | Branch to one of two target blocks. | Cannot be used to implement a non-terminating loop.     |
```

Comment:
What does "cannot be used" mean? As noted above, this is unenforceable in the general case.